### PR TITLE
docs: add tidb roadmap

### PR DESCRIPTION
**Pull request by ran-huang**
### What problem does this PR solve?

Add a 1-year roadmap for tidb.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    <td>Non-prepared Plan Cache</td>
+    <td>Support plan cache for general SQL statements in a session to save cache resources, improve the hit rate of general execution plans, and improve SQL performance.</td>
+    <td>Non-prepared plan cache. Improve real-time and throughputs of OLTP in general scenarios, save PoC time, and increase PoC win rate.</td>
+    <td>January 2023</td>
+  </tr>
+  <tr>
+    <td>SQL blocklist</td>
+    <td>Support a rule-based SQL blocklist mechanism.</td>
+    <td>In multi-service aggregation scenarios, provide SQL management and control capabilities, and improve cluster stability by prohibiting high-resource-consuming SQL statements.</td>
+    <td>2022.02</td>
```
Comment (qiancai) [inline roadmap.md:86]:
```suggestion
    <td>February 2023</td>
```

Comment (qiancai) [reply roadmap.md:86]: 2022 should be 2023?

File: roadmap.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    <th>Description</th>
+    <th>Estimated Time of Delivery</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td rowspan="2">Backup and restore</td>
+    <td>EBS snapshot-based backup</td>
+    <td>BR provides an MVP version of EBS snapshot-based backup and restore.</td>
+    <td>2022.9</td>
```
Comment (qiancai) [inline roadmap.md:152]:
```suggestion
    <td>September 2022</td>
```

File: roadmap.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    <td>October 2022</td>
+  </tr>
+  <tr>
+    <td rowspan="2">Password complexity check</td>
+    <td>A strong password is required.</td>
+    <td>To improve security, empty passwords and weak passwords are not allowed.<br>The required password length is not less than 8. The password must contain an uppercase letter, a lowercase letter, a number, and a character.</td>
+    <td>November 2022</td>
+  </tr>
+  <tr>
+    <td>TiDB provides password expiration management and requires users to change passwords regularly.</td>
```
Comment (qiancai) [inline roadmap.md:215]: The scenario for this row is "TiDB支持密码有效期管理" instead of Password complexity check.

File: roadmap.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+<tbody>
+  <tr>
+    <td rowspan="2">Support JSON</td>
+    <td>Support JSON function</td>
+    <td>In business scenarios that require flexible schema definitions (such as SaaS, Web3, and gaming), the application can use JSON to store information for ODS, transaction indicators, commodities, game characters, and props.</td>
+    <td>September 2022</td>
+  </tr>
+  <tr>
+    <td><ul><li>Support expression indexes</li><li>Support generated columns</li></ul></td>
+    <td>Provides query acceleration for specific field indexes in JSON scenarios.</td>
```
Comment (qiancai) [inline roadmap.md:27]:
```suggestion
    <td>Provide query acceleration for specific field indexes in JSON scenarios.</td>
```

File: roadmap.md (hunk 3 above)
Comment (ran-huang) [reply roadmap.md:226]: added

File: roadmap.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+  <tr>
+    <td>DDL parallel framework</td>
+    <td>Implement a single DDL parallel execution framework for DDL acceleration</td>
+    <td>Accelerate DDL in scenarios such as creating indexes for large tables, modifying a column's default value, and modifying a column type.</td>
+    <td>January 2023</td>
+  </tr>
+  <tr>
+    <td>Non-prepared Plan Cache</td>
+    <td>Support plan cache for general SQL statements in a session to save cache resources, improve the hit rate of general execution plans, and improve SQL performance.</td>
+    <td>Non-prepared plan cache. Improve real-time and throughputs of OLTP in general scenarios, save PoC time, and increase PoC win rate.</td>
```
Comment (qiancai) [inline roadmap.md:83]: @easonn7 . Please help check if we need to mention "increase PoC win rate" here.

File: roadmap.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+  <tr>
+    <td>Support replicating data to object storage such as S3.</td>
+    <td>TiCDC supports replicating data changes to common object storage services.</td>
+    <td>November 2022</td>
+  </tr>
+  <tr>
+    <td>Data migration</td>
+    <td>TiDB Lightning supports table-level and partition-level incremental data import.</td>
+    <td>TiDB Lightning provides comprehensive table-level and partition-level data import capabilities</td>
+    <td>December 2022</td>
```
Comment (ran-huang) [inline roadmap.md:184]: @qiancai PTAL

File: roadmap.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+  </tr>
+  <tr>
+    <td>PB-level scalability</td>
+    <td>Support dynamic region size adjustment (heterogeneous).</td>
+    <td>For scenarios with fast business growth and a large amount of data.</td>
+    <td>June 2023</td>
+  </tr>
+  <tr>
+    <td>Instance plan cache</td>
+    <td>Support cross-session plan cache, save cache resources, improve the hit rate of general execution plans, and improve SQL performance</td>
```
Comment (qiancai) [inline roadmap.md:112]:
```suggestion
    <td>Support cross-session plan cache, save cache resources, improve the hit rate of general execution plans, and improve SQL performance.</td>
```

File: roadmap.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+  </tr>
+  <tr>
+    <td>Password expiration</td>
+    <td>TiDB provides password expiration management and requires users to change passwords regularly.</td>
+    <td>Reduce the security risk of password cracking or leakage caused by using the same password for a long time.</td>
+    <td>December 2022</td>
+  </tr>
+  <tr>
+    <td>Password policy management</td>
+    <td>TiDB provides a password reuse mechanism and brute-force cracking prevention capabilities</td>
```
Comment (lilin90) [inline roadmap.md:232]: Added a period for complete sentences to keep the style consistent.
```suggestion
    <td>TiDB provides a password reuse mechanism and brute-force cracking prevention capabilities.</td>
```

File: roadmap.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+    <td>TiDB supports password policy management to protect password security.</td>
+    <td>March 2023</td>
+  </tr>
+  <tr>
+    <td>Column-level access control</td>
+    <td>TiDB supports column-level privilege management</td>
```
Comment (lilin90) [inline roadmap.md:238]:
```suggestion
    <td>TiDB supports column-level privilege management.</td>
```

File: roadmap.md [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+    <td>TiDB already supports cluster-level, database-level, and table-level privilege management. On top of that, TiDB will support column-level privilege management to meet the principle of least privilege and provide fine-grained data access control.</td>
+    <td>June 2023</td>
+  </tr>
+  <tr>
+    <td>Audit logging capability refactor</td>
+    <td>Support configurable audit log policies, configurable audit filters (filter by objects, users, and operation types), and visual access to audit logs</td>
```
Comment (lilin90) [inline roadmap.md:244]:
```suggestion
    <td>Support configurable audit log policies, configurable audit filters (filter by objects, users, and operation types), and visual access to audit logs.</td>
```

File: roadmap.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+  </tr>
+  <tr>
+    <td>Support replicating data to object storage such as S3.</td>
+    <td>TiCDC supports replicating data changes to common object storage services.</td>
+    <td>November 2022</td>
+  </tr>
+  <tr>
+    <td>Data migration</td>
+    <td>TiDB Lightning supports table-level and partition-level incremental data import.</td>
+    <td>TiDB Lightning provides comprehensive table-level and partition-level data import capabilities</td>
```
Comment (qiancai) [inline roadmap.md:183]:
```suggestion
    <td>TiDB Lightning provides comprehensive table-level and partition-level data import capabilities.</td>
```

File: roadmap.md (hunk 6 above)
Comment (qiancai) [reply roadmap.md:184]: Done.

File: roadmap.md (hunk 8 above)
Comment (ran-huang) [reply roadmap.md]: done in [749ad0e](https://github.com/pingcap/tidb/pull/38002/commits/749ad0e640d7eadd43b00f857886f46bb14c5914)

**Comment by allengaoo** [comment]
👍

**Review by kaixu120811 — APPROVED** [review]
LGTM

**Review by ljun0712 — APPROVED** [review]

**Comment by jiezhu2007** [comment]
LGTM

File: roadmap.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+</thead>
+<tbody>
+  <tr>
+    <td rowspan="2">Backup and restore</td>
+    <td>AWS EBS snapshot-based backup</td>
+    <td>BR provides an MVP version of AWS EBS snapshot-based backup and restore.</td>
+    <td>September 2022</td>
+  </tr>
+  <tr>
+    <td>Backup and restore based on AWS EBS or GCP Persistent Disk snapshot</td>
```
Comment (qiancai) [inline roadmap.md:157]:
```suggestion
    <td>Backup and restore based on AWS EBS or GCP persistent disk snapshots</td>
```

Comment (qiancai) [reply roadmap.md:157]: See https://cloud.google.com/compute/docs/disks/snapshots

**Review by qiancai — APPROVED** [review]

**Review by IANTHEREAL — APPROVED** [review]

**Pull request merged**